### DISCUSSION TITLE
Confusion over S" and SLITERAL

### DISCUSSION THREAD

**Discussion by SirWumpus** [ForthHub/discussion#128] [Q&A] [answered]
The Forth 200x Draft 19.1 for `S"` defines both interpretation and compilation semantics, while 17.6.1.2212 SLITERAL is compile-only.  The Rationale A.17.6.1.2212 SLITERAL suggests that it can be used to define 6.1.2165 S" .
```
: S" ( "ccc<quote>" -- )
     [CHAR] " PARSE POSTPONE SLITERAL
; IMMEDIATE 
````
Does this not imply that `SLITERAL` must also perform the `S"` interpretation semantics too? 

This appears to be an oversight between the Forth 2012 standard and the proposed Draft.  In the standard `S"` (and `S\"`) and `SLITERAL` have compile-only semantics and so work together nicely, while in the Draft `S"` (and `S\"`) now have interpretation semantics, but failed to update `SLITERAL` and its Rationale to address the change.

Is this a failure (bug) in the proposed Draft or am I missing something with respect to `SLITERAL` implementation?

**Comment by ruv** [comment, marked as answer]
> Does this not imply that `SLITERAL` must also perform the `S"` interpretation semantics too?

This example implementation was correct in Forth-2012, and it just become incorrect in the new draft.

There are two versions for the word `s"` in Forth-2012 — CORE ([6.1.2165](https://forth-standard.org/standard/core/Sq)) and FILE ([11.6.1.2165](https://forth-standard.org/standard/file/Sq)). And a difference is that interpretation semantics are undefined in the CORE  version.

The rationale [A.17.6.1.2212](https://forth-standard.org/standard/rationale#rat:string:SLITERAL) provides an example for the CORE version, it says: "The current functionality of [6.1.2165 `S"`](https://forth-standard.org/standard/core/Sq) may be provided by the following definition". So, it provides an example for the CORE version of `s"`, that cannot be used in interpretation state by a standard program.

On the standard meeting in 2017 it [was decided](https://github.com/Forth-Standard/forth200x/blob/master/meetings/2017-notes#L92) to remove the FILE version of `s"`, and update the CORE version of `s"` (move from FILE).

> This appears to be an oversight between the Forth 2012 standard and the proposed Draft.

Yes, you are right — it's an oversight.

> am I missing something with respect to `SLITERAL` implementation?

No. `sliteral` was not changed.
